akd/src/append_only_zks.rs

```diff
@@ -573,6 +577,35 @@ impl Azks {
         self.latest_epoch = epoch;
     }
 
+    /// Gets the sibling node of the passed node's child in the "opposite" of the passed direction.
+    async fn get_sibling_node<S: Database + Sync + Send>(
```

Comment:
Thanks for extracting this logic! Much cleaner :)

Reply:
Now that I think of it, is it worth adding test coverage over this specifically?